packages/web/src/components/search/SearchBox.test.js

```diff
@@ -660,3 +405,194 @@ it('should render SearchBox with featured suggestions', () => {
+				<SearchBox
+					testMode
+					componentId="MockSearchBox"
+					dataField="original_title"
+					mockData={{
+						hits: AI.MOCK_HITS_DATA,
+						AI_RESPONSE: AI.MOCK_AI_RESPONSE,
+						rawData: AI.MOCK_RAW_DATA,
+					}}
+					enableAI
+					isOpen
+				/>
+			</ReactiveBase>,
+		);
+
+	// Wait for all updates to complete
+	await act(async () => {
+		await new Promise(resolve => setTimeout(resolve, 0));
+	});
+
+	expect(elem.toJSON()).toMatchSnapshot();
+});
+
+
+it('should render AI response & askButton & enterButton', async () => {
+	const elem = renderer
+		.create(
+			<ReactiveBase app="test" url="https://a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61@localhost:800">
+				<SearchBox
+					testMode
+					componentId="MockSearchBox"
+					dataField="original_title"
+					mockData={{
+						hits: AI.MOCK_HITS_DATA,
+						AI_RESPONSE: AI.MOCK_AI_RESPONSE,
+						rawData: AI.MOCK_RAW_DATA,
+					}}
+					AIUIConfig={{
+						askButton: true,
+					}}
+					enableAI
+					isOpen
+				/>
+			</ReactiveBase>,
+		);
+
+	// Wait for all updates to complete
+	await act(async () => {
+		await new Promise(resolve => setTimeout(resolve, 0));
+	});
+
+	expect(elem.toJSON()).toMatchSnapshot();
+});
+
+
+it('should render AI response & triggerOn=question', async () => {
+	const elem = renderer
+		.create(
+			<ReactiveBase app="test" url="https://a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61@localhost:800">
+				<SearchBox
+					componentId="MockSearchBox"
+					dataField="original_title"
+					defaultValue="Harry potter?"
+					mockData={{
+						hits: AI.MOCK_HITS_DATA,
+						AI_RESPONSE: AI.MOCK_AI_RESPONSE,
+						rawData: AI.MOCK_RAW_DATA,
+					}}
+					AIUIConfig={{
+						askButton: true,
+						triggerOn: 'question',
+					}}
+					enableAI
+					isOpen
+				/>
+			</ReactiveBase>,
+		);
+
+	// Wait for all updates to complete
+	await act(async () => {
+		await new Promise(resolve => setTimeout(resolve, 0));
+	});
+
+	expect(elem.toJSON()).toMatchSnapshot();
+});
+
+it('should render AI response & triggerOn=manual', async () => {
+	const elem = renderer
+		.create(
+			<ReactiveBase app="test" url="https://a03a1cb71321:75b6603d-9456-4a5a-af6b-a487b309eb61@localhost:800">
+				<SearchBox
+					componentId="MockSearchBox"
+					dataField="original_title"
+					defaultValue="Harry potter?"
+					mockData={{
+						hits: AI.MOCK_HITS_DATA,
+						AI_RESPONSE: AI.MOCK_AI_RESPONSE,
+						rawData: AI.MOCK_RAW_DATA,
+					}}
+					AIUIConfig={{
+						askButton: true,
+						triggerOn: 'manual',
+						renderTriggerMessage: 'Click to trigger AI 🤖',
+					}}
+					enableAI
+					isOpen
+				/>
+			</ReactiveBase>,
+		);
+
+	// Wait for all updates to complete
+	await act(async () => {
+		await new Promise(resolve => setTimeout(resolve, 0));
+	});
+
+	expect(elem.toJSON()).toMatchSnapshot();
+});
+
+
+it('should render CUSTOM AI response ', async () => {
```

Comment:
@SavvyShah Please check the snapshot generated for this test, the snapshot doesn't represent the expected result.

Reply:
OK. Fixed.